pages/content/posts/localizing-your-website.md

```diff
@@ -0,0 +1,33 @@
+---
+author: "Vinyl Da.i'gyu-Kazotetsu"
+title: "Why it is important to localize your website"
+date: "2022-12-14"
+tags: ["development"]
+ShowToc: false
+ShowBreadCrumbs: false
+excerpt: "A client of mine had recently asked me why it was so important to localize their web app. They asked me, why spend money on localization when you can do it for free with Google Translate? There is so much to unpack about why that approach and mindset is wrong."
+---
+
+A site owner recently asked me why it was so important to localize their web app.  They told me: “we just plug Google Translate in and it’s been fine. Why spend hundreds, thousands, tens of thousands on localization when you can do it for free with Google Translate?”
+
+There is _so_ much to unpack about why that approach and mindset _doesn't work_.
+
+There are many considerations when it comes to being inclusive of diverse audiences, whether it’s to accommodate sensory impairments, theming (dark/light mode), localization, and more. When designing your website or software, or even just updating it, it's important to consider all aspects of inclusion every step of the way to engage as wide of an audience as possible. It is important not to overlook anything either, as then you risk your product appearing broken or incomplete.
+
+Take the `alt` attribute text for images as an example.  Let’s say you have ten images, and you have properly and accurately described nine of the images by including appropriate `alt` text values (or marked them as decorative, but either way, you have followed accessibility standards for 90% of your images).  If a user who benefits from and relies upon image descriptions, such as users with cognitive differences and visually impaired screen reader users, then goes to the page containing images without `alt` text, then suddenly, the element, important content, cannot be accessed.
+	
+Is the image just decoration? Does the image contain critical information that is essential to the rest of the page?  The screen reader has no way of knowing and the user doesn’t know.  To the visually impaired user, the web page page is incomplete; broken even.
+
+The same can be said for localization.  Now, in this case, there is already very powerful machine translation software (such as [Google Translate](https://translate.google.com/)) that is capable of translating entire documents, webpages, and even oral conversations, and these translation capabilities are only going to get more powerful.  However, automated services can only do so much: the ones that are readily available provide translations only; they cannot understand context, including user interface options, idioms, or other special phrases such as product names, etc.
```

Comment:
I don't think we should say "applications" because there are also services and APIs for machine translation accessible through other apps and websites that may not have their own app.

Reply:
mostly, i wanted to change the `is` to `are` and it sounded weird b/c software is singular and plural but sounds singular only

Reply:
Would "solutions" instead of "software" sound reasonable?

Reply:
i removed the applications. Commit if ok with my suggestion